lib/Parse/ParseType.cpp

```diff
@@ -1610,7 +1624,9 @@ bool Parser::canParseType() {
   case tok::kw__:
     consumeToken();
     break;
-
+  case tok::integer_literal:
+    consumeToken();
+    break;
```

Comment:
Does this mean we also return `true` for `1.Protocol`, `1 & 1`, `1 -> Int` and what else is being checked below. Shouldn’t we return immediately from here?

Reply:
Similarly, these are also diagnosed during type checking:
```
/Users/alejandro.alonso/6.swift:1:8: error: integer unexpectedly used in a type position
1 | let x: 1.Type
  |        `- error: integer unexpectedly used in a type position
2 | let y: -1 & 2

/Users/alejandro.alonso/6.swift:2:9: error: integer unexpectedly used in a type position
1 | let x: 1.Type
2 | let y: -1 & 2
  |         `- error: integer unexpectedly used in a type position
```